Makefile

```diff
@@ -925,3 +924,100 @@ ui-development: build/api build/api-db build/local-api-data-watcher-pusher build
+	curl -sSLo local-dev/kind https://github.com/kubernetes-sigs/kind/releases/download/$(KIND_VERSION)/kind-$(ARCH)-amd64
+	chmod a+x local-dev/kind
+endif
+
+local-dev/jq:
+ifeq ($(GOJQ_VERSION), $(shell jq -v 2>/dev/null | sed -nE 's/gojq ([0-9.]+).*/v\1/p'))
+	$(info linking local jq version $(KIND_VERSION))
+	ln -s $(shell command -v jq) ./local-dev/jq
+else
+	$(info downloading gojq version $(GOJQ_VERSION) for $(ARCH))
+	curl -sSL https://github.com/itchyny/gojq/releases/download/$(GOJQ_VERSION)/gojq_$(GOJQ_VERSION)_$(ARCH)_amd64.tar.gz | tar -xzC local-dev --strip-components=1 gojq_$(GOJQ_VERSION)_$(ARCH)_amd64/gojq
+	mv ./local-dev/{go,}jq && chmod a+x local-dev/jq
+endif
+
+.PHONY: helm/repos
+helm/repos: local-dev/helm
+	# install repo dependencies required by the charts
+	./local-dev/helm repo add harbor https://helm.goharbor.io
+	./local-dev/helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx
+	./local-dev/helm repo add stable https://charts.helm.sh/stable
+	./local-dev/helm repo add bitnami https://charts.bitnami.com/bitnami
+
+.PHONY: kind/cluster
+kind/cluster: local-dev/kind
+	# these IPs are a result of the docker config on our build nodes
+	export KUBECONFIG=$$(mktemp) \
+		KINDCONFIG=$$(mktemp -p . kindconfig.XXX) \
+		&& chmod 644 $$KUBECONFIG \
+		&& curl -sSLo $$KINDCONFIG https://raw.githubusercontent.com/uselagoon/lagoon-charts/$(CHARTS_TREEISH)/test-suite.kind-config.yaml \
+		&& echo '  [plugins."io.containerd.grpc.v1.cri".registry.configs."registry.192.168.48.2.nip.io:32443".tls]'  >> $$KINDCONFIG \
+		&& echo '    insecure_skip_verify = true'                                                                    >> $$KINDCONFIG \
+		&& echo '  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."registry.192.168.48.2.nip.io:32080"]'      >> $$KINDCONFIG \
+		&& echo '    endpoint = ["http://registry.192.168.48.2.nip.io:32080"]'                                       >> $$KINDCONFIG \
+		&& echo '  [plugins."io.containerd.grpc.v1.cri".registry.configs."registry.192.168.160.2.nip.io:32443".tls]' >> $$KINDCONFIG \
+		&& echo '    insecure_skip_verify = true'                                                                    >> $$KINDCONFIG \
+		&& echo '  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."registry.192.168.160.2.nip.io:32080"]'     >> $$KINDCONFIG \
+		&& echo '    endpoint = ["http://registry.192.168.160.2.nip.io:32080"]'                                      >> $$KINDCONFIG \
+		&& echo '  [plugins."io.containerd.grpc.v1.cri".registry.configs."registry.172.25.0.2.nip.io:32443".tls]'    >> $$KINDCONFIG \
+		&& echo '    insecure_skip_verify = true'                                                                    >> $$KINDCONFIG \
+		&& echo '  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."registry.172.25.0.2.nip.io:32080"]'        >> $$KINDCONFIG \
+		&& echo '    endpoint = ["http://registry.172.25.0.2.nip.io:32080"]'                                         >> $$KINDCONFIG \
```

Comment:
These entries are _only_ for allowing containerd to pull images from the registry insecurely. e.g. after building a drupal image, pushing it to the local registry, and defining a deployment.

The mirror is a good idea! I didn't add it initially since I was only testing in CI where we are logged in. I'll add it.

Reply:
>  The different private IPs are the ones that docker provides on our CI nodes. It's not the cleanest.

gotcha, maybe add a comment or so there, so this is more clearer why they are there?

Reply:
I figured out how to calculate this instead of hard-coding IPs.